src/main/java/Echo.java

```diff
@@ -0,0 +1,131 @@
+                    return list.unmark(index) + ui.line();
+                } catch (NumberFormatException e) {
+                    return "Please enter a valid task number\n" + ui.line();
+                }
+            }
+        case "find":
+            try {
+                return list.find(parts[1]).printList() + ui.line();
+            } catch (StringIndexOutOfBoundsException e) {
+                return "No such items in the list!\n" + ui.line();
+            }
+
+        case "todo":
+            try {
+                Task toDoTask = parser.parseToDoTask(parts[1]);
+                list.addTask(toDoTask);
+                return ui.affirm() + toDoTask.getDescription() + "\n" + ui.line() +
+                        String.format("Now you have %d tasks in the list\n", list.size());
+            } catch (ArrayIndexOutOfBoundsException e) {
+                return "Oh no! Please input a ToDo description!";
+            }
+
+        case "deadline":
+            try {
+                Task deadlineTask = parser.parseDeadlineTask(parts[1]);
+                list.addTask(deadlineTask);
+                return ui.affirm() + deadlineTask.getDescription() +
+                                "\n" + ui.line() + String.format("Now you have %d tasks in the list\n", list.size());
+            } catch (StringIndexOutOfBoundsException e) {
+                return "Incorrect format of adding deadline tasks. " +
+                        "Use '/by to specify the deadline after the task description";
+            } catch (DateTimeParseException e) {
+                return "Please input the correct deadline format /by yyyy-MM-dd XXXX <- Time";
+            }
+
+        case "event":
+            try {
+                Task eventTask = parser.parseEventTask(parts[1]);
+                list.addTask(eventTask);
+
+                return ui.affirm() + eventTask.getDescription() +
+                        "\n" + ui.line() + String.format("Now you have %d tasks in the list\n", list.size());
+            } catch (StringIndexOutOfBoundsException e) {
+                return "Incorrect format of adding event tasks. " +
+                        "Use '/from to specify the start and /to to specify the end " +
+                        "after the task description\n";
+            } catch (DateTimeParseException e) {
+                return "Incorrect format of adding event tasks timings. /from yyyy-MM-dd XXXX <- Time" +
+                        "and /to yyyy-MM-dd XXXX <- Time\n";
+            }
+
+        case "delete":
+            int index = Integer.parseInt(parts[1]) - 1;
+            return list.removeTask(index);
+
+        case "bye":
+            try {
+                storage.saveTasksToFile(this.list);
+                return ui.goodbye();
+            } catch (IOException e) {
+                return "Unable to save chat data to local disk!\n";
+            }
+        default:
+            return ui.invalidCommand();
+        }
+    }
+}
```

Comment:
Well written. Clear program flow structure that adheres to OOP principles. Exceptions are caught early on, and dealt with well.